### ISSUE TITLE
[BUG] Can't get actual test result info in onStepEnd 

### ISSUE THREAD

**Issue by vovsemenv** [microsoft/playwright#14364]
**Context:**
- Playwright Version: 1.22.2
- Operating System: Mac
- Node.js version: 16
- Browser: Chromium
- Extra: [any specific details about your environment]
- 
**Code Snippet**

I have custom reporter like this
```javascript
class CustomReporter {
    onStepEnd(_test, _result, step) {
        console.log(_result.attachments)   
    }
    onTestEnd(_test, _result, step) {
        console.log(_result.attachments)   
    }
}
```
and test like this
```javascript
import { test } from '@playwright/test';
test('testing attachments', async ({}, testInfo) => {
    testInfo.attach('4', {body:'444'})
});
```

output
```
❯ npm t

> new-project@1.0.0 test
> playwright test

[]
[]
[
  {
    name: '4',
    path: undefined,
    contentType: 'text/plain',
    body: <Buffer 34 34 34>
  }
]
```

**Describe the bug**
in onStepEnd hook i have `test` but when try to check current attachments always get empty array and attachments appear only in onTestEnd step.

**Comment by yury-s** [comment]
Currently the attachments are only passed to the reporter in `onTestEnd`. Can you tell us more about the use case where you need them in onStepEnd ?

**Comment by vovsemenv** [comment]
yes, we use attachments to provide extra information about tests in allure reporter and want to start parsing it when step ends.

**Comment by vovsemenv** [comment]
@yury-s sorry for pinging you. I think a lot of people find out this feature really useful. Did you have any plans  about it?

**Comment by ramam-v** [comment]
@yury-s Will this feature be supported? I work in a regulated environment  where  we have to capture screenshots at step level to meet the regulatory requirements. This feature will be of great help.

**Comment by wtclark** [comment]
Agreed, capturing screenshots at the step level is important and helpful for troubleshooting test failures. This functionality exists in other reporting libraries so it's not an uncommon practice.
